md2conf/api.py

```diff
@@ -212,17 +218,36 @@ def upload_attachment(
 
         url = self._build_url(path)
 
-        with open(attachment_path, "rb") as attachment_file:
+        if not raw_data:
+            with open(attachment_path, "rb") as attachment_file:
+                file_to_upload = {
+                    "comment": comment,
+                    "file": (
+                        attachment_name,  # will truncate path component
+                        attachment_file,
+                        content_type,
+                        {"Expires": "0"},
+                    ),
+                }
+                LOGGER.info("Uploading attachment: %s", attachment_name)
+                response = self.session.post(
+                    url,
+                    files=file_to_upload,  # type: ignore
+                    headers={"X-Atlassian-Token": "no-check"},
+                )
```

Comment:
Much of this looks like a repetition of the block below. Perhaps we might want to encapsulate this into a function?